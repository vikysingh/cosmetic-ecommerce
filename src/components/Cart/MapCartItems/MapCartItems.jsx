import React, { useState } from 'react'

import styles from "./MapCartItems.module.css"
import globalStyles from "../../../styles/Globals.module.css"

import CartCard from "../CartCard/CartCard"

import { removeFromCart } from "../../../redux/actions/cart/cartActionGenerators"
import { connect } from "react-redux"

function Mapcartitems({ cart, dispatch }) {

    const [ items, setItems ] = useState([])

    React.useEffect(() => {
        mapData()
    }, [])
    
    function mapData() {

        cart.map(item => {
            fetch(`https://makeup-api.herokuapp.com/api/v1/products/${item.id}.json`)
            .then(response => response.json().then(data => {
                setItems(items => [...items, { imgUrl: data.api_featured_image, name: data.name, 
                    price: data.price, id: data.id }])
            }))
        })
    }
    return (
        <div id={styles.MAP_CART_ITEMS} className={globalStyles.flexColumnCenterCenter}>
            {
                items.length === 0 ? <h4>Loading</h4> :
                items.map(cartItem => <CartCard imgUrl={cartItem.imgUrl} key={cartItem.id}
                name={cartItem.name} quantity={cart[items.indexOf(cartItem)].quantity} 
                id={cartItem.id} price={cartItem.price} dispatch={dispatch} action={{ removeFromCart }}
                 />)
            }
        </div>
    )
}

const mapStateToProps = state => state

export default React.memo(connect(mapStateToProps)(Mapcartitems))