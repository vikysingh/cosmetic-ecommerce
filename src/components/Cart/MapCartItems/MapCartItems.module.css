#MAP_CART_ITEMS {
    width: 100%;
    height: 100%;
    /* padding: 5% 0; */
    justify-content: flex-start;
}