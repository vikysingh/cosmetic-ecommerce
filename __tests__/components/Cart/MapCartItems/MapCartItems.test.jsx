import React from "react"
import { shallow } from "enzyme"

import MapCartItems from "../../../../components/Cart/MapCartItems/MapCartItems"
import { dummyCartData, dummyCartProductData } from "../../../../test_files/dummyTestData"

test('Should render MapCartItems from Cart', () => {
    const wrapper = shallow(<MapCartItems productList={dummyCartProductData}
        dispatch={dummyCartData.dispatch} />)

    expect(wrapper).toMatchSnapshot()
})