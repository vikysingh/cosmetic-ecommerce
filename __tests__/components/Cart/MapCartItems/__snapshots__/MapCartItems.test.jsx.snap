// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MapCartItems from Cart 1`] = `
<div
  className="flexColumnCenterCenter"
  id="MAP_CART_ITEMS"
>
  <[object Object]
    action={
      Object {
        "removeFromCart": [Function],
      }
    }
    dispatch={[Function]}
    id={49}
    imgUrl="some url"
    key="49"
    name="Product name"
    price="14.99"
    quantity={4}
  />
  <[object Object]
    action={
      Object {
        "removeFromCart": [Function],
      }
    }
    dispatch={[Function]}
    id={23}
    imgUrl="some url"
    key="23"
    name="Product name"
    price="14.99"
    quantity={4}
  />
</div>
`;
